DirectProgramming/DPC++FPGA/Tutorials/Features/private_copies/src/CMakeLists.txt

```diff
@@ -0,0 +1,81 @@
+# To see a Makefile equivalent of this build system:
+# https://github.com/oneapi-src/oneAPI-samples/blob/master/DirectProgramming/DPC++/ProjectTemplates/makefile-fpga
+
+set(SOURCE_FILE private_copies.cpp)
+set(TARGET_NAME private_copies)
+set(EMULATOR_TARGET ${TARGET_NAME}.fpga_emu)
+set(FPGA_TARGET ${TARGET_NAME}.fpga)
+
+# FPGA board selection
+if(NOT DEFINED FPGA_BOARD)
+    set(FPGA_BOARD "intel_a10gx_pac:pac_a10")
+    message(STATUS "FPGA_BOARD was not specified.\
+                    \nConfiguring the design to run on the default FPGA board ${FPGA_BOARD} (Intel(R) PAC with Intel Arria(R) 10 GX FPGA). \
+                    \nPlease refer to the README for information on board selection.")
+else()
+    message(STATUS "Configuring the design to run on FPGA board ${FPGA_BOARD}")
+endif()
+
+# This is a Windows-specific flag that enables exception handling in host code
+if(WIN32)
+    set(WIN_FLAG "/EHsc")
+endif()
+
+# A DPC++ ahead-of-time (AoT) compile processes the device code in two stages.
+# 1. The "compile" stage compiles the device code to an intermediate representation (SPIR-V).
+# 2. The "link" stage invokes the compiler's FPGA backend before linking.
+#    For this reason, FPGA backend flags must be passed as link flags in CMake.
+set(EMULATOR_COMPILE_FLAGS "${WIN_FLAG} -fintelfpga -DFPGA_EMULATOR")
+set(EMULATOR_LINK_FLAGS "-fintelfpga")
+set(AOC_SEED_FLAG "-Xsseed=4 -Xsparallel=2") # Special flags passed to to Quartus
+set(HARDWARE_COMPILE_FLAGS "${WIN_FLAG} -fintelfpga")
+if(FPGA_BOARD STREQUAL "intel_s10sx_pac:pac_s10")
+    # hyper-optimized-handshaking applies to only S10 and Agilex

```

Comment:
1/ why do you need to disable loop handshaking for this sample?
2/ if you really need it, please modify by doing something like:

```
# Hyper-optimization is turned off because XYZ
if(FPGA_BOARD MATCHES ".*a10.*")
    set(HANDSHAKING "")
else()
    set(HANDSHAKING "-Xshyper-optimized-handshaking=off")
endif()

...
set(EMULATOR_COMPILE_FLAGS "${WIN_FLAG} -fintelfpga -DFPGA_EMULATOR")
set(EMULATOR_LINK_FLAGS "-fintelfpga")
set(HARDWARE_COMPILE_FLAGS "${WIN_FLAG} -fintelfpga")
set(HARDWARE_LINK_FLAGS "-fintelfpga -Xshardware ${HANDSHAKING} -Xsboard=${FPGA_BOARD} ${USER_HARDWARE_FLAGS}")
```